File: 백준/Silver/10828. 스택/스택.py
import sys

n = int(sys.stdin.readline())

stack = []

for _ in range(n):
    order = sys.stdin.readline().rstrip()
    if order == "pop":
        if stack:
            print(stack.pop())
        else:
            print(-1)
    elif order == "size":
        print(len(stack))
    elif order == "empty":
        if stack:
            print(0)
        else:
            print(1)
    elif order == "top":
        if stack:
            print(stack[-1])
        else:
            print(-1)
    else:
        push, num = order.split()
        stack.append(int(num))

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <deque>
#include <string>

using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);
	
	int n;
	cin >> n;

	deque<int> dq;
	string cmd;
	for (int i = 0; i < n; i++) {
		cin >> cmd;
		if (cmd == "push") {
			int num;
			cin >> num;
			dq.push_back(num);
		}
		else if (cmd == "pop") {
			if (!dq.empty()) {
				cout << dq.back() << "\n";
				dq.pop_back();
			}
			else {
				cout << -1 << "\n";
			}
		}
		else if (cmd == "size") {
			cout << dq.size() << "\n";
		}
		else if (cmd == "empty") {
			if (dq.empty()) {
				cout << 1 << "\n";
			}
			else {
				cout << 0 << "\n";
			}
		}
		else if (cmd == "top") {
			if (!dq.empty()) {
				cout << dq.back() << "\n";
			}
			else {
				cout << -1 << "\n";
			}
		}
	}

	return 0;
}